src/transaction-builders/ERC20TransactionBuilder.ts

```diff
@@ -1,25 +1,29 @@
-import { Contract, BigNumber } from 'ethers'
+import { BigNumber, utils } from 'ethers'
 import { RawTransactionBuilder } from './RawTransactionBuilder'
 import { Safe, SafeTransaction } from '@gnosis.pm/safe-core-sdk'
 
 export class ERC20TransactionBuilder extends RawTransactionBuilder {
-  #erc20Contract!: Contract
+  public erc20Address: string
+  #erc20Contract: utils.Interface
 
-  constructor(safe: Safe, erc20Contract: Contract) {
+  constructor(safe: Safe, erc20Address: string) {
```

Comment:
I would add a check to validate the contract address, maybe something similar to [this](https://github.com/rsksmart/safe-factory-sdk/blob/449b22f15ba920326dd3d95d6a3b802c851abfe9/src/contracts.ts#L31).